lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -152,8 +152,12 @@ def call
 
           # convert the includes with virtual attributes to includes with proper associations
           records_by_assoc = records.group_by { |rec| assoc_cache[rec.class] }
-          # if these are the same includes, then do the preloader work
-          return super if records_by_assoc.size == 1 && records_by_assoc.keys.first == associations
+          # If the association were already translated, then short circuit / do the standard preloader work.
+          # When replace_virtual_fields removes the outer array, match that too.
+          if records_by_assoc.size == 1 &&
+            (associations == records_by_assoc.keys.first || associations == [records_by_assoc.keys.first])
```

Comment:
Would flatten or similar help keep it consistent for comparison?

Reply:
The `associaions` will probably always be an array. Even a hash becomes a single element array with a hash in it.

`fix_association` optimizes the array right out of there.
So we end up comparing `:symbol` with `[:symbol]`, they are different and we get one extra set of lambdas (and one extra `group_by`) - all with the same class.

This is an optimization only, Without this change, all works fine.

Reply:
Right, would `Array.wrap(...)` normalize to an array make this less complex or introduce a possible performance hit for the case when it's already an array?

Reply:
I had thought I could avoid this via:

```diff
- assoc_cache = Hash.new { |h, klass| h[klass] = klass.replace_virtual_fields(associations) }
+ assoc_cache = Hash.new { |h, klass| h[klass] = Array.wrap(klass.replace_virtual_fields(associations)) }
```

Since we have  `Array.wrap` on that field later in the block, but ended up with an infinite loop.

So if we do this, I'll need to do the reverse:

```ruby
records_by_assoc.keys.first == [associations] || records_by_assoc.keys.first == associations
```

Still tempting